### ISSUE TITLE
propMap in subcommand throws UnknownOption exception

### ISSUE THREAD

**Issue by stevedlawrence** [scallop/scallop#39]
When trying to get the propMap of a subcommand, scallop throws an UnknownOption exception. For example, assume we have this:

```
object Conf extends ScallopConf(List("sub", "-Dkey1=value1", "key2=value2")) {
    val sub = new scallop.Subcommand("sub") {
        val properties = props[String]('D', keyName="key", valueName="value", descr="key/value map in subcommand")
    }
}
```

The following two statements give the expected output:

```
> Conf.sub.properties("key1")
Some("value1")
> Conf.sub.properties("key2")
Some("value2")
```

However, this throws an exeption:

```
> Conf.sub.propMap[String]('D')
Exception in thread "main" org.rogach.scallop.exceptions.UnknownOption: Unknown option 'D'
        at org.rogach.scallop.Scallop$$anonfun$get$6.apply(Scallop.scala:488)
        at org.rogach.scallop.Scallop$$anonfun$get$6.apply(Scallop.scala:488)
        at scala.Option.getOrElse(Option.scala:108)
        at org.rogach.scallop.Scallop.get(Scallop.scala:488)
        at org.rogach.scallop.Scallop.apply(Scallop.scala:499)
        at org.rogach.scallop.ScallopConf.propMap(ScallopConf.scala:323)
        at daffodil.Main$.main(Main.scala:109)
        at daffodil.Main.main(Main.scala
```

**Comment by Rogach** [comment]
Yes, that's bad. The problem here is that those maps do not follow the behavior of plain options, thus escaping the net of type system. The actual name of the option includes the subcommand name, so you can get that map using the following incantation:

``` scala
Conf.sub.rootConfig.builder("sub\0D")(implicitly[Manifest[Map[String,String]]])
```

Maybe it will be better if I will just change the return type of `props` and `propsLong` to return a map? What do you think?

**Comment by stevedlawrence** [comment]
Changing the return type of props and propsLong to a Map is actually much more intuitive to me. That's actually what I was expecting when I started using props, and had to go to the documentation to find that I had to use propMap to get what I wanted. I'm in favor of the change.

**Comment by Rogach** [comment]
I implemented the fix and published a snapshot (for scala 2.9.2):

``` scala
resolvers += "sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "org.rogach" %% "scallop" % "0.7.1~1~SNAPSHOT"
```

**Comment by stevedlawrence** [comment]
Works perfectly for me. Thanks!

**Comment by Rogach** [comment]
I just published version `0.8.0`, that includes the change. It would be synced to Maven Central in 1-2 hours.
